crates/cheatnet/tests/common/assertions.rs

```diff
@@ -33,3 +35,18 @@ pub fn assert_error(call_contract_output: CallResult, expected_data: impl Into<S
         if msg == expected_data.into(),
     ));
 }
+
+pub trait ClassHashAssert {
+    fn success_class_hash(self) -> ClassHash;
+}
+
+impl ClassHashAssert for DeclareResult {
+    fn success_class_hash(self) -> ClassHash {
+        match self {
+            DeclareResult::Success(class_hash) => class_hash,
+            DeclareResult::AlreadyDeclared(class_hash) => {
+                panic!("Class hash {class_hash} is already declared")
+            }
+        }
+    }
```

Comment:
This is confusing to me. At least, if you want to code it this way, name it in a way that suggests that this function is panicking/asserting something. E.g. `unwrap_success_class_hash`. 

At a caller site, it's not obvious what is it doing at all.

Reply:
Or we could change the `DeclareResult` a bit, so it returns a struct and then implement something along the lines of `unwrap_success. 

Then we could write `declare(state, contract_name, &contracts_data).unwrap_success().class_hash`.

Reply:
I like the name `unwrap_success`, but I don't understand why it should return a struct. We should just return the hash, not a struct containing it. The `DeclareResult` is a container. The same behavior is seen in Rust's `Result` enum. When you call `unwrap_err`, you get the inner value, not a struct containing it. So, I think it should just be `declare(state, contract_name, &contracts_data).unwrap_success()`.

Reply:
`unwrap_success` is okay to me, the idea with returning a struct was just an different option, but agreed that it might not make that much sense.